conventional REMP2 and OREMP2 (#2653)
psi4/psi4: doc/sphinxman/source/occ.rst

This final equation corresponds to the usual Newton-Raphson step.

* OREMP

> The REMP hybrid perturbation theory is a constrined mixture of the |MollerPlesset| perturbation theory and the

Reviewer: ```suggestion
The REMP hybrid perturbation theory is a constrained mixture of the |MollerPlesset| perturbation theory and the
```

Reviewer 2: Me bad, thanks for spotting.

Retaining the Excitation degree perturbation theory([Fink:2006:RE]_, [Behnle:2019:REMP]_).
The mixing ratio is determined by the parameter :math':`A`:

.. math::
   \widehat{H}^{(0)}_\text{REMP} = (1-A)\widehat{H}^{(0)}_\text{RE} + A\widehat{H}^{(0)}_\text{MP}

Technically, the second order of RE corresponds to LCCD for RHF and UHF references. REMP2 and its orbital-optimized variant OREMP2

> are thus straightforward to implement in a (O)LCCD program by appropriate scaling of residuum vector contributions and density matrices.

Reviewer: ```suggestion
are thus straightforward to implement in a (O)LCCD program by appropriate scaling of residual vector contributions and density matrices.
```

Starting in v1.4, MP2.5 and MP3 default to the density-fit algorithm. Set |globals__mp_type| to ``CONV`` to get previous behavior.

> Publications resulting from the use of the non-OO CC codes should cite the following publications:

Reviewer: ```suggestion
Publications resulting from the use of the REMP codes should cite the following publications:
```

I'm very confused why this said "non-OO CC" originally. Am I missing something?

Author: yes, there's a list of all the other methods and citations that I didn't pick into this PR. Probably best to consider docs as a whole later on.

    +                         +--------------------------------------------------------------+----------------------+----------------------+---------------------------+
    |                         | Cholesky-Decomposed Lambda-CCSD(T)                           | RHF                  | ---                  | |globals__cc_type| CD     |
    +-------------------------+--------------------------------------------------------------+----------------------+----------------------+---------------------------+

>     | remp2                   | 2nd order REMP hybrid perturbation theory                    | RHF/UHF              | ---                  | |globals__cc_type| CONV   |

Reviewer: _Technically_ we don't have density-fitted yet. If we're on track to get that in by 1.7, I leave it up to you whether you want this part of the docs to be accurate in the meantime.
